internal/service/redshiftserverless/snapshot_copy_configuration.go

```diff
@@ -0,0 +1,243 @@
+		return
+	}
+
+	conn := r.Meta().RedshiftServerlessClient(ctx)
+
+	output, err := findSnapshotCopyConfigurationByID(ctx, conn, data.NamespaceName.ValueString(), data.ID.ValueString())
+
+	if tfresource.NotFound(err) {
+		response.Diagnostics.Append(fwdiag.NewResourceNotFoundWarningDiagnostic(err))
+		response.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		response.Diagnostics.AddError(fmt.Sprintf("reading Redshift Serverless Snapshot Copy Configuration (%s)", data.ID.ValueString()), err.Error())
+		return
+	}
+
+	response.Diagnostics.Append(fwflex.Flatten(ctx, output, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *resourceSnapshotCopyConfiguration) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var new, old snapshotCopyConfigurationResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &old)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	response.Diagnostics.Append(request.Plan.Get(ctx, &new)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().RedshiftServerlessClient(ctx)
+
+	input := &redshiftserverless.UpdateSnapshotCopyConfigurationInput{}
+	response.Diagnostics.Append(fwflex.Expand(ctx, new, input)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	input.SnapshotCopyConfigurationId = aws.String(new.ID.ValueString())
+
+	_, err := conn.UpdateSnapshotCopyConfiguration(ctx, input)
+
+	if err != nil {
+		response.Diagnostics.AddError(fmt.Sprintf("updating Redshift Serverless Snapshot Copy Configuration (%s)", new.ID.ValueString()), err.Error())
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &new)...)
+}
+
+func (r *resourceSnapshotCopyConfiguration) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	var data snapshotCopyConfigurationResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().RedshiftServerlessClient(ctx)
+
+	_, err := conn.DeleteSnapshotCopyConfiguration(ctx, &redshiftserverless.DeleteSnapshotCopyConfigurationInput{
+		SnapshotCopyConfigurationId: aws.String(data.ID.ValueString()),
+	})
+	if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		return
+	}
+	if err != nil {
+		response.Diagnostics.AddError(fmt.Sprintf("deleting Redshift Serverless Snapshot Copy Configuration (%s)", data.ID.ValueString()), err.Error())
+		return
+	}
+}
+
+func findSnapshotCopyConfigurationByID(ctx context.Context, conn *redshiftserverless.Client, namespaceName string, ID string) (*awstypes.SnapshotCopyConfiguration, error) {
+	input := &redshiftserverless.ListSnapshotCopyConfigurationsInput{
+		NamespaceName: aws.String(namespaceName),
+		MaxResults:    aws.Int32(100),
+	}
+
+	var configuration *awstypes.SnapshotCopyConfiguration
+
+	for {
```

Comment:
You can use `redshiftserverless.NewListSnapshotCopyConfigurationsPaginator()` to paginate with the SDK rather than doing it manually using `NextToken`.

Reply:
Refactored as suggested. Please let me know if that's what you had in mind @drewtul. Cheers!